error[E0593]: function is expected to take 0 arguments, but it takes 1 argument
  --> $DIR/callable.rs:10:12
   |
LL |     test::<fn(i32) -> i32>();
   |            ^^^^^^^^^^^^^^ expected function that takes 0 arguments
   |
note: required by a bound in `test`
  --> $DIR/callable.rs:7:12
   |
LL | fn test<T: std::ops::Callable<()>>() {}
   |            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `test`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0593`.
